docs/get-details/dapps/smart-contracts/apps/index.md

```diff
@@ -661,7 +727,278 @@ The `int 0` represents the current application and `int 1` would reference the f
+
+Each entry in the box array allows access to only 1kb of data. For example, if a box is sized to 4kb, the transaction must use four entries in this array. To claim an allotted entry a corresponding app Id and box name need to be added to the box ref array. If you need more than the 1kb associated with that specific box name, you can either specify the box ref entry more than once or, preferably, add “empty” box refs `[0,””]` into the array. If you specify 0 as the app Id the box ref is for the application being called. 
+
+For example, suppose the contract needs to read “BoxA” which is 1.5kb, and “Box B” which is 2.5kb, this would require four entries in the box ref array and would look something like:
+
+```py
+boxes=[[0, "BoxA"],[0,"BoxB"], [0,""],[0,""]]
+``` 
+
+The box reference budget is based on the sizes of the boxes accessed, not the amount of data read or written. For example, if a contract accesses “Box A” with a size of 2kb and “Box B” with a size of 10 bytes, this requires both boxes be in the box reference array and one additional reference, which should be an “empty” box reference. 
+
+Access budgets are summed across multiple application calls in the same transaction group. For example in a group of two smart contract calls, there is room for 16 array entries (8 per app call), allowing access to 16kb of data. If an application needs to access a 16kb box named “Box A”, it will need to be grouped with one additional application call and the box reference array for each transaction in the group should look similar to this:
+
+Transaction 0: [0,”Box A”],[0,””],[0,””],[0,””],[0,””],[0,””],[0,””],[0,””]
+Transaction 1: [0,””],[0,””],[0,””],[0,””],[0,””],[0,””],[0,””],[0,””]
+
+Box refs can be added to the boxes array using `goal` or any of the SDKs.
+
+=== "Goal"
+    ```goal
+    goal app method --app-id=53 --method="add_member2()void" --box="53,str:BoxA" --from=CONP4XZSXVZYA7PGYH7426OCAROGQPBTWBUD2334KPEAZIHY7ZRR653AFY
+    ```
+
+=== "Python SDK"
+    ```py
+    # Algorand Python SDK
+    # Using ATC
+    atc = AtomicTransactionComposer()
+    atc.add_method_call(app_id, my_method, addr, sp, signer, method_args=[1,5], boxes=[[app_id, “key”]],
+    )
+    ```
+
+=== "Beaker"
+    ```py
+    #Beaker framework
+    result = app_client.call(
+        Myapp.my_method,
+    boxes=[[app_client.app_id, "key"]],
+    )
+    ```
+## Creating a Box
+The AVM supports two opcodes `box_create` and `box_put` that can be used to create a box. 
+The `box_create` opcode takes two parameters, the name and the size in bytes for the created box. The `box_put` opcode takes two parameters as well. The first parameter is the name and the second is a byte array to write. Because the AVM limits byte arrays to 4,096, `box_put` can only be used for boxes with length <= 4,096.
```

Comment:
We should try to be consistent about specifying byte sizes (i.e. 4096 if we mean 4kb)

```suggestion
The `box_create` opcode takes two parameters, the name and the size in bytes for the created box. The `box_put` opcode takes two parameters as well. The first parameter is the name and the second is a byte array to write. Because the AVM limits any element on the stack to 4kb (#TODO link to stack parameter), `box_put` can only be used for boxes with length <= 4kb.
```

Reply:
also, need to add the link to the parameter table here